### ISSUE TITLE
Make bookend async

### ISSUE THREAD

**Issue by jithine** [screwdriver-cd/screwdriver#1477]
**What happened**:

Coverage bookend can take lots of time depending on project size. It would have been nice to have a feature to proceed rest of the processing of the pipeline while coverage bookend takes it's time to complete.

**What you expected to happen**:

Provide a mechanism to mark coverage bookend to either
1. Run async, ie start it in background as soon as the dependent step which produces the test result are complete.
    1. This can be generalized to run any steps in parallel.
2. Continue processing of rest of build and not wait for coverage step to complete.
    1. This can be generalized to make any step behave this way. 

**How to reproduce it**:

Run coverage bookend on a large pipeline which takes lot of time for Sonar scanner to complete.

**Comment by yoshwata** [comment]
I hope that it will be adjustable that each bookend runs async or not. 
Some pipelines in our cluster would be pretty fast if scm bookend and cache bookend runs with async.
![1e7ce880-c906-11e9-95ef-04fb4332e11b](https://user-images.githubusercontent.com/32473622/63997155-9f232b00-cb38-11e9-9d01-51434494deb7.png)
